core/embed/rust/src/ui/component/placed.rs

```diff
@@ -121,3 +121,135 @@ where
         d.close();
     }
 }
+
+pub struct Floating<T> {
+    inner: T,
+    size: Offset,
+    border: Offset,
+    align: Alignment2D,
+}
+
+impl<T> Floating<T> {
+    pub const fn new(size: Offset, border: Offset, align: Alignment2D, inner: T) -> Self {
+        Self {
+            inner,
+            size,
+            border,
+            align,
+        }
+    }
+
+    pub const fn top_right(side: i16, border: i16, inner: T) -> Self {
+        let size = Offset::uniform(side);
+        let border = Offset::uniform(border);
+        Self::new(size, border, TOP_RIGHT, inner)
+    }
+}
+
+impl<T> Component for Floating<T>
+where
+    T: Component,
+{
+    type Msg = T::Msg;
+
+    fn place(&mut self, bounds: Rect) -> Rect {
+        let mut border = self.border;
+        let area = match self.align.0 {
+            Alignment::Start => bounds.split_left(self.size.x).0,
+            Alignment::Center => panic!("alignment not supported"),
+            Alignment::End => {
+                border.x = -border.x;
+                bounds.split_right(self.size.x).1
+            }
+        };
+        let area = match self.align.1 {
+            Alignment::Start => area.split_top(self.size.y).0,
+            Alignment::Center => panic!("alignment not supported"),
+            Alignment::End => {
+                border.y = -border.y;
+                area.split_bottom(self.size.y).1
+            }
+        };
+        self.inner.place(area.translate(border))
+    }
+
+    fn event(&mut self, ctx: &mut EventCtx, event: Event) -> Option<Self::Msg> {
+        self.inner.event(ctx, event)
+    }
+
+    fn paint(&mut self) {
+        self.inner.paint()
+    }
+}
+
+#[cfg(feature = "ui_debug")]
+impl<T> crate::trace::Trace for Floating<T>
+where
+    T: Component,
+    T: crate::trace::Trace,
+{
+    fn trace(&self, d: &mut dyn crate::trace::Tracer) {
+        d.open("Floating");
+        d.field("inner", &self.inner);
+        d.close();
+    }
+}
+
+pub struct VSplit<T, U> {
```

Comment:
Thea meaning of `V` in `VSplit` is `vertical`? I also wonder what is the difference between creating two buttons this way versus the "normal" way - cannot we unite it?

Reply:
Yes. The usual way is to specify a ratio of button sizes, while here the caller specifies exact size  of (one of) the buttons. The button placement will need to be a bit overhauled for the updated style, and it might end up unified as you suggest, preferably in next PR.